src/NHibernate.Test/OneToOneType/Fixture.cs

```diff
@@ -0,0 +1,61 @@
+﻿using NHibernate.Test.NHSpecificTest;
+using NUnit.Framework;
+
+namespace NHibernate.Test.OneToOneType
+{
+	[TestFixture]
+	public class Fixture : BugTestCase
+	{
+		protected override void OnTearDown()
+		{
+			using (var s = Sfi.OpenSession())
+			using (var tx = s.BeginTransaction())
+			{
+				s.CreateQuery("delete from Details").ExecuteUpdate();
+				s.CreateQuery("delete from Owner").ExecuteUpdate();
+
+				tx.Commit();
+			}
+		}
+
+		[Test]
+		public void OneToOnePersistedOnOwnerUpdate()
```

Comment:
Can you also add test that checks that cache is properly reset when `owner.Details`  is changed from not-null to null.